tests/terminal.rs

```diff
@@ -47,3 +50,96 @@ fn terminal_draw_returns_the_completed_frame() -> Result<(), Box<dyn Error>> {
     assert_eq!(frame.area, Rect::new(0, 0, 8, 8));
     Ok(())
 }
+
+#[test]
+fn terminal_insert_before_moves_viewport() -> Result<(), Box<dyn Error>> {
```

Comment:
I've now cleaned up these tests using your suggestions. I've also added some descriptions which hopefully showcase the behaviour they are trying to ensure.

I've also added a comment to address the following in `terminal_insert_before_moves_viewport` specifically because it happened to be the first although this may be something worth adding to the docstring?

> It's not clear to me the effect of calling insert_before after rather than before the draw call - I'd suspect that most of the > time we have a buffer that ha been drawn to before and after the call?